code/__HELPERS/_logging.dm

```diff
@@ -113,8 +113,13 @@
 		WRITE_LOG(GLOB.world_attack_log, "ATTACK: [text]")
 
 /proc/log_manifest(ckey, datum/mind/mind,mob/body, latejoin = FALSE)
-	if (CONFIG_GET(flag/log_manifest))
-		WRITE_LOG(GLOB.world_manifest_log, "[ckey] \\ [body.real_name] \\ [mind.assigned_role] \\ [mind.special_role ? mind.special_role : "NONE"] \\ [latejoin ? "LATEJOIN":"ROUNDSTART"]")
+    if (CONFIG_GET(flag/log_manifest))
+        if(istype(body, /mob/living/carbon))
+            var/mob/living/carbon/M = body
+            if(M.dna.species)
+                WRITE_LOG(GLOB.world_manifest_log, "[ckey] \\ [body.real_name] \\ [mind.assigned_role] \\ [mind.special_role ? mind.special_role : "NONE"] \\ [latejoin ? "LATEJOIN":"ROUNDSTART"] \\ [format_text(M.dna.species.name)]")
```

Comment:
and if their species is null? or dna is null?

Reply:
There